extern/sector-storage/stores/index.go

```diff
@@ -51,15 +51,33 @@ type SectorStorageInfo struct {
 	Primary bool
 }
 
+// SectorIndex is the Miner's sector store Index/Registry that keeps track of all storage directories
+// and sector files in the sector stores of all Workers.
+// The Miner exposes this Index API over JSON RPC to all remote workers. Local workers can of-course call it from within the process.
 type SectorIndex interface { // part of storage-miner api
+	// StorageAttach is called by workers to register a new storage directory or update information about an existing storage directory
+	// with the Index.
 	StorageAttach(context.Context, StorageInfo, fsutil.FsStat) error
+	// StorageInfo is used to get detailed information about a storage directory with the given ID.
 	StorageInfo(context.Context, ID) (StorageInfo, error)
+	// StorageReportHealth is called by workers to report the health of a registered storage directory.
+	// This is the mechanism my which the Index tracks healthy/unhealthy/available/unavailable storage directories across workers.
 	StorageReportHealth(context.Context, ID, HealthReport) error
 
+	// StorageDeclareSector is called by workers to register new sector files or update information about
+	// existing sector files with the Index.
+	// `primary` is set to true if the sector files are in the long term storage of the worker.
+	// `primary` is set to false if the sector files are in the sealing scratch space of the worker.
```

Comment:
This reads imperative and expresses usage. It doesn't explain what is the function of `primary`. What does it stand for?

```
`primary` specifies the nature of storage. If true, it indicates the sector files are in the long-term storage of the worker. Otherwise, it indicates they're in its scatch space.
```

Reply:
`primary` means the "main" copy of the sector, and can be true for sector files still in stealing storage.

This is mostly used for cleanup purposes - the `store.RemoveCopies` method will look at sector files, and if there is a primary sector file - remove the non-primary files (normally there shouldn't be any stray sector files, but copies can be left in sealing paths when things get shutdown forcefully)